internal/pkg/service/buffer/worker/service/slice_watch.go

```diff
@@ -0,0 +1,137 @@
+package service
+
+import (
+	"context"
+	"sync"
+
+	gonanoid "github.com/matoous/go-nanoid/v2"
+	etcd "go.etcd.io/etcd/client/v3"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/schema"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/slicestate"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/prefixtree"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+type activeSlicesWatcher struct {
+	// slices contains keys of all slices in these states: opened, closing, uploading, failed
+	// The slice key is removed when the uploaded state is reached.
```

Comment:
V tomto commite je implementovane a otestovane cakanie kym sa uploadnu vsetky slices.
 - file v state `closing`, musi pockat, kym su vsetky slices v stave `uploaded`, az tak sa moze prepnut file do stavu `importing` a tym sa spusti import.

Reply:
Tu je iba to cakanie.